Detectors/Vertexing/src/SVertexer.cxx

```diff
@@ -421,10 +421,10 @@ bool SVertexer::checkV0(const TrackCand& seedP, const TrackCand& seedN, int iP,
   if (checkForCascade) {
     int nCascAdded = 0;
     if (hypCheckStatus[HypV0::Lambda] || !mSVParams->checkCascadeHypothesis) {
-      nCascAdded += checkCascades(rv0, pV0, p2V0, iN, NEG, ithread);
+      nCascAdded += checkCascades(rv0, pV0, p2V0, iN, NEG, vlist, ithread);
     }
     if (hypCheckStatus[HypV0::AntiLambda] || !mSVParams->checkCascadeHypothesis) {
-      nCascAdded += checkCascades(rv0, pV0, p2V0, iP, POS, ithread);
+      nCascAdded += checkCascades(rv0, pV0, p2V0, iP, POS, vlist, ithread);
```

Comment:
alternatively, one can add V0s not only for the vertex with the best cosPA value but for each PV with respect to which cosPA is acceptable, then to look for the cascades in old way, for this PV only.

Reply:
but this in principle could leave us with many duplicates that are not even associated with a cascade, right? If what I am saying is correct maybe the first strategy is the best one, and I will just duplicate the V0 if its corresponding cascade is found